app/controllers/public_health_controller.rb

```diff
@@ -120,83 +131,164 @@ def self_reporting
 
   protected
 
-  def filter_sort_paginate(params, data)
-    # Filter on search
-    filtered = filter(params, data)
+  def patients_by_type(workflow, tab)
+    if tab == :transferred_in
+      patients = current_user.jurisdiction.transferred_in_patients.where(isolation: workflow == :isolation)
+    elsif tab == :transferred_out
+      patients = current_user.jurisdiction.transferred_out_patients.where(isolation: workflow == :isolation)
+    else
+      patients = current_user.viewable_patients
 
-    # Sort on columns
-    sorted = sort(params, filtered)
+      if workflow == :exposure
```

Comment:
`return patients.method if condition`? that way it doesn't have to fall through the other `if` statements?

Why do these get overwritten a lot? Only 1 tab param can be given at a time, correct?

Reply:
looks much cleaner now, thanks for the suggestion! https://github.com/SaraAlert/SaraAlert/commit/7f06c26894cd5271163119a8b04b909943f4bd46#diff-f332c4e8d749e2f0c02c18036420ef37R116